src/neo4j_graphrag/experimental/pipeline/config/config_parser.py

```diff
@@ -0,0 +1,739 @@
+        return param.resolve(self._global_data)
+
+    def resolve_params(self, params: dict[str, ParamConfig]) -> dict[str, Any]:
+        """Resolve all parameters
+
+        Returning dict[str, Any] because parameters can be anything (str, float, list, dict...)
+        """
+        return {
+            param_name: self.resolve_param(param)
+            for param_name, param in params.items()
+        }
+
+    @abc.abstractmethod
+    def parse(self, resolved_data: dict[str, Any] | None = None) -> Any:
+        raise NotImplementedError()
+
+
+T = TypeVar("T")
+"""Generic type to help (a bit) mypy with the return type of the parse method"""
+
+
+class ObjectConfig(AbstractConfig, Generic[T]):
+    """A config class to represent an object from a class name
+    and its constructor parameters.
+    """
+
+    class_: str | None = Field(default=None, validate_default=True)
+    """Path to class to be instantiated."""
+    params_: dict[str, ParamConfig] = {}
+    """Initialization parameters."""
+
+    DEFAULT_MODULE: ClassVar[str] = "."
+    """Default module to import the class from."""
+    INTERFACE: ClassVar[type] = object
+    """Constraint on the class (must be a subclass of)."""
+    REQUIRED_PARAMS: ClassVar[list[str]] = []
+    """List of required parameters for this object constructor."""
+
+    @field_validator("params_")
+    @classmethod
+    def validate_params(cls, params_: dict[str, Any]) -> dict[str, Any]:
+        """Make sure all required parameters are provided."""
+        for p in cls.REQUIRED_PARAMS:
+            if p not in params_:
+                raise ValueError(f"Missing parameter {p}")
+        return params_
+
+    def get_module(self) -> str:
+        return self.DEFAULT_MODULE
+
+    def get_interface(self) -> type:
+        return self.INTERFACE
+
+    def parse(self, resolved_data: dict[str, Any] | None = None) -> T:
+        """Import `class_`, resolve `params_` and instantiate object."""
+        self._global_data = resolved_data or {}
+        if self.class_ is None:
+            raise ValueError("`class_` is not defined")
+        klass = self._get_class(self.class_, self.get_module())
+        if not issubclass(klass, self.get_interface()):
+            raise ValueError(
+                f"Invalid class '{klass}'. Expected a subclass of '{self.get_interface()}'"
+            )
+        params = self.resolve_params(self.params_)
+        try:
+            obj = klass(**params)
+        except TypeError as e:
+            raise e
+        return obj  # type: ignore[return-value]
+
+
+class Neo4jDriverConfig(ObjectConfig[neo4j.Driver]):
+    REQUIRED_PARAMS = ["uri", "user", "password"]
+
+    @field_validator("class_", mode="before")
+    @classmethod
+    def validate_class(cls, class_: Any) -> str:
+        """`class_` parameter is not used because we're always using the sync driver."""
+        if class_:
+            # logger.info("Parameter class_ is not used")
```

Comment:
Should this be removed?

Reply:
I actually think it should be there (ie not commented), WDYT?

Reply:
I'm unclear on how this class gets used, does it always return `"not used"`?

From the name I expected to return a `bool`

Reply:
It's a pydantic field validator, returning the validated field. In the case of the neo4j driver, we do not allow the class to be configured, that's why I think the logger could be useful. We could also have two separate base classes: one for neo4j (without the `class_` attribute) and one for the other types (LLM, embedders) with the `class_` attribute, but I was thinking it's already quite a complex structure.